lib/json-schema/schema/reader.rb

```diff
@@ -15,7 +14,34 @@ class ReadRefused < StandardError
       def initialize(location, type)
         @location = location
         @type = type
-        super("Read of #{type == :uri ? 'URI' : type} at #{location} refused!")
+        super(error_message)
+      end
+
+      private
+
+      def error_message
```

Comment:
That is never read from the base class or did I overlook something? Wouldn't it be reasonable to require each subclass to implement a custom error message?